pkg/restapi/kms/operation/operations.go

```diff
@@ -304,6 +310,62 @@ func (o *Operation) decryptHandler(rw http.ResponseWriter, req *http.Request) {
 	})
 }
 
+func (o *Operation) computeMACHandler(rw http.ResponseWriter, req *http.Request) {
+	var request computeMACReq
+	if ok := parseRequest(&request, rw, req); !ok {
+		return
+	}
+
+	keystoreID := mux.Vars(req)[keystoreIDQueryParam]
+	keyID := mux.Vars(req)[keyIDQueryParam]
+
+	provider := prepareKMSProvider(rw, o.provider, keystoreID, request.Passphrase)
+	if provider == nil {
+		return
+	}
+
+	srv := kmsservice.NewService(provider)
+	mac, err := srv.ComputeMAC(keystoreID, keyID, []byte(request.Data))
+	if err != nil {
+		writeErrorResponse(rw, http.StatusInternalServerError, fmt.Sprintf(computeMACFailure, err))
```

Comment:
do not leak error details: trustbloc/hub-kms#38

Reply:
working on this now, I decided to not put these changes into the current PR given that there is a ticket for refactoring error handling